flycheck.el

```diff
@@ -7016,6 +7017,39 @@ See URL `http://hadolint.lukasmartinelli.ch/'."
      (flycheck-remove-error-file-names "/dev/stdin" errors)))
   :modes dockerfile-mode)
 
+(defun flycheck-credo--working-directory (&rest _ignored)
+  "Find directory with from which we can run credo."
+  (and buffer-file-name
+       (locate-dominating-file buffer-file-name "deps/credo")))
+
+(flycheck-def-option-var flycheck-elixir-credo-strict nil elixir-credo
+  "Enable strict mode in credo.
+When non-nil, pass the `--strict' flag to credo."
+  :type 'boolean
+  :safe #'booleanp
+  :package-version '(flycheck . "0.32"))
+
+(flycheck-define-checker elixir-credo
+  "Elixir credo checker."
+  :command ("mix"
+            "credo"
+            (option-flag "--strict" flycheck-elixir-credo-strict)
+            "--format"
+            "flycheck"
+            "--read-from-stdin"
+            source-original)
+  :standard-input t
+  :predicate
+  (lambda () (and buffer-file-name
+                  (locate-dominating-file buffer-file-name "deps/credo")))
```

Comment:
Isn't that the same as `flycheck-credo--working-directory`?
It should likely be an `enabled` property, too

Reply:
When do you prefer `:enabled` over `:predicate`?
Probably has something to do with this:?
> This property behaves as `:predicate', except that it's only
     called the first time a syntax checker is to be used in a buffer.

Reply:
Correct — there's no need to check for that file on each run.

Reply:
Okay, I've changed it. This now means that you'd have to reopen a buffer after installing credo into a project to active the checker, right?